### ISSUE TITLE
Unhandled JSON parse exception when handling malformed responses from ContentStack API

### ISSUE THREAD

**Issue by vitorsdcs** [contentstack/contentstack-java#128]
The SDK seems to be throwing an unhandled exception when ContentStack API returns a non-JSON response.

The following error is thrown during execution of [CSHttpConnection.java#L264](https://github.com/contentstack/contentstack-java/blob/master/src/main/java/com/contentstack/sdk/CSHttpConnection.java#L264):

`JSONException: A JSONObject text must begin with '{' at 0 [character 1 line 1]`

In case of malformed responses from ContentStack API, I would expect the ResultCallback to be invoked and contain details about the error, instead of an exception being thrown.

**Comment by reeshika-h** [comment]
Hey @vitorsdcs thanks for raising this, we will look into the issue .

cc: @cs-raj @netrajpatel @shafeeqd959

**Comment by reeshika-h** [comment]
Hey @vitorsdcs kindly check out the latest version. Do let us know if you face any further issues.

**Comment by goetas** [comment]
Hi, I work on the same team as @vitorsdcs , the issue is still happening on the 2.0.0 release. Did not upgrade to 2.0.1 due to https://github.com/contentstack/contentstack-java/issues/148

**Comment by reeshika-h** [comment]
Hi @goetas! Just a heads-up that we'll be releasing the fixes for the critical bug mentioned in #148. This issue (#128) has been addressed in version 2.0.1. We'll keep you posted.

**Comment by reeshika-h** [comment]
Hey @vitorsdcs, could you please let us know if this has been resolved?

**Comment by goetas** [comment]
@reeshika-h yes, after `[v2.0.2](https://github.com/contentstack/contentstack-java/tree/v2.0.2)` the issue is not happening anymore
